File: Medium/49. Group Anagrams/solution.cs
public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        Dictionary<string, List<string>> myDict = new Dictionary<string, List<string>>();
        foreach (string str in strs){
            var sortedString = new string(str.OrderBy(c => c).ToArray());
            if (!myDict.ContainsKey(sortedString)){
                myDict.Add(sortedString, new List<string> {str});
            }
            else{
                myDict[sortedString].Add(str);
            }
        }
        Console.WriteLine("here");
        var result = new List<IList<string>>();
        foreach(var kvp in myDict){
            result.Add(kvp.Value);
        }
        return result;
    }
}

File: Medium/49. Group Anagrams/solution.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string> > myMap;
        for (string s : strs){
            string temp = s;
            sort(temp.begin(), temp.end());
            myMap[temp].push_back(s);
        }
        vector<vector<string> > result;
        for (auto it : myMap){
            result.push_back(it.second);
        }
        return result;
    }
};